src/components/Group.tsx

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+
+import EmployeeList from "./EmployeeList";
+import { EmployeeInterface } from "./Employee";
+import { GroupType } from "./GroupList";
+import Loading from "./Loading";
+import fetchEmployees from "../utils/fetchEmployees";
+
+const Group = (): JSX.Element => {
+  const [employees, setEmployees] = useState<EmployeeInterface[]>([]);
+  const [loading, setLoading] = useState<boolean>(true);
+  const params = useParams();
+  const groupType: GroupType = params.groupType;
+  const group: string | undefined = params.group;
+
+  useEffect(() => {
+    setLoading(true);
+    fetchEmployees(
+      setEmployees,
+      setLoading,
+      (tempList: Array<EmployeeInterface>): Array<EmployeeInterface> | null => {
+        if (!groupType || !group) {
+          return null;
+        }
+
+        const matches: Array<EmployeeInterface> | null = [];
+
+        for (const employee of tempList) {
+          if (groupType === "titles") {
+            if (employee.jobTitle === group) {
+              matches.push(employee);
+            }
+          } else if (groupType === "locations") {
+            if (employee.location === group) {
+              matches.push(employee);
+            }
+          }
+        }
+
+        return matches;
+      }
+    );
+  }, [group, groupType]);
```

Comment:
This is not how this should be. `fetchEmployees` should only return data or an error in case of an error. `setEmployees` and `setLoading` should be called outside this function.

Example of how this should be:
```js

  useEffect(() => {
    const fetchData = async () => {
      setLoading(true);
      const employees = fetchEmployees();
      setLoading(false);
      // Do whatever you want with the data here
      setEmployees(employees);
    };
    fetchData();
  }, []);
  ```